File: dao/food-dao.js
const db = require('../config/db');

class FoodDAO {

    async getFoods(){
        let results = await db.query(`SELECT * FROM public.food `).
        catch(console.log);
        return results.rows;
    };

    async getFoodbyID(id){
        let results = await db.query(`SELECT * FROM FOOD WHERE id = $1`,[id]).
        catch(console.log);
        return results.rows[0];
    };

    async getFoodsbyRestaurant(restaurantID){
        let results = await db.query(`SELECT * FROM FOOD WHERE restaurantID = $1`,[restaurantID]).
        catch(console.log);
        return results.rows;
    };

    async createFood(allergens, description, price, name, category, restaurantID, img){
        await db.query(`INSERT INTO FOOD (allergens, description, price, name, category, restaurantID, img) VALUES ($1, $2, $3, $4, $5, $6, $7)`,[allergens, description, price, name, category, restaurantID, img])
            .catch(console.log);
        return;
    };

    async updateFood(id, allergens, description, price, name, category, restaurantID, img){
        await db.query(`UPDATE FOOD SET allergens = $1, description = $2, price = $3, name = $4, category = $5, restaurantID = $6, img = $7 WHERE id = $8`,[allergens, description, price, name, category, restaurantID, img, parseInt(id)])
            .catch(console.log);

        return;
    };

    async deleteFood(id){
        await db.query(`DELETE FROM FOOD WHERE id=$1`,[parseInt(id)])
            .catch(console.log);

        return;
    };

};

module.exports = FoodDAO;

File: MasodikMaven/src/main/java/application/dao/FoodDAO.java
package application.dao;

import java.util.ArrayList;
import java.util.Map;
import java.util.List;
import javax.annotation.PostConstruct;
import javax.sql.DataSource;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.jdbc.core.support.JdbcDaoSupport;
import org.springframework.stereotype.Repository;

import application.model.Food;

@Repository
public class FoodDAO extends JdbcDaoSupport  {

    @Autowired
    DataSource dataSource;

    @PostConstruct
    private void initialize(){
        setDataSource(dataSource);
    }

    public void insertFood(Food food) {
        String sql = "INSERT INTO food(name) VALUES (?)";
        getJdbcTemplate().update(sql, new Object[]{
                food.getName()
        });
    }

    public List<Food> listFoods(){
        String sql = "SELECT * FROM food";
        List<Map<String, Object>> rows = getJdbcTemplate().queryForList(sql);

        List<Food> result = new ArrayList<Food>();
        for(Map<String, Object> row:rows){
            Food food = new Food();
            food.setId((Integer)row.get("id"));
            food.setName((String)row.get("name"));
            result.add(food);
        }

        return result;
    }

    public Food getFoodbyID(int id){
        String sql = "SELECT * FROM food WHERE id="+id;
        List<Map<String, Object>> rows = getJdbcTemplate().queryForList(sql);

        List<Food> result = new ArrayList<Food>();
        for(Map<String, Object> row:rows){
            Food food = new Food();
            food.setId((Integer)row.get("id"));
            food.setName((String)row.get("name"));
            result.add(food);
        }

        return result.get(0);
    }

    public void deleteFood(int id){
        String sql = "DELETE FROM food WHERE id="+id;
        getJdbcTemplate().update(sql);
    }

    public void updateFood(int id, String name){
        String sql = "UPDATE food SET name='"+name+"' WHERE id="+id;
        getJdbcTemplate().update(sql);
    }

}